src/puppetlabs/puppetdb/utils.clj

```diff
@@ -264,6 +264,55 @@
       (apply assoc m new-kvs)
       m)))
 
+(defn str-last
```

Comment:
My guess is this will use reflection, but if you use `clojure.string/starts-with?` and `clojure.string/ends-with?` you probably won't need it anymore.